feat: add C implementation for `stats/base/dists/planck/mgf`

yuvimittal opened the pull request:
Resolves #4944

## Description

> What is the purpose of this pull request?

This pull request:

- Add C implementation for @stdlib/stats/base/dists/planck/mgf

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #4944 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

yuvimittal committed:
Add C implementation for @stdlib/stats/base/dists/planck/mgf

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

yuvimittal committed:
Refactor random number generation in JS benchmarks

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

stdlib-bot committed:
chore: update copyright years

stdlib-bot committed:
chore: update copyright years

yuvimittal commented:
@kgryte , i think it is ready to merge .

Planeshifter committed:
fix: address review issues in `stats/base/dists/planck/mgf` C implementation

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/planck/mgf/benchmark/benchmark.native.js:
```diff
@@ -0,0 +1,70 @@
+
+// MAIN //
+
+bench( pkg + '::native', opts, function benchmark( b ) {
```
```suggestion
bench( pkg+'::native', opts, function benchmark( b ) {
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/planck/mgf/benchmark/benchmark.native.js:
```diff
@@ -0,0 +1,70 @@
+	lambda = new Float64Array( len );
+	for ( i = 0; i < len; i++ ) {
+		t[ i ] = uniform(0.0, 10.0);
+		lambda[ i ] = uniform(0.1, 10.0);
```
```suggestion
		t[ i ] = uniform( 0.0, 10.0 );
		lambda[ i ] = uniform( 0.1, 10.0 );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/planck/mgf/examples/c/example.c:
```diff
@@ -0,0 +1,41 @@
+}
+
+int main( void ) {
+	double t;
+	double lambda;
+	double y;
+	int i;
```
```suggestion
	double lambda;
	double t;
	double y;
	int i;
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/planck/mgf/examples/c/example.c:
```diff
@@ -0,0 +1,41 @@
+		t = random_uniform( 0.0, 10.0 );
+		lambda = random_uniform( 0.1, 5.0 );
+		y = stdlib_base_dists_planck_mgf( t, lambda );
+		printf( "t: %lf, lambda: %lf, M_X(t; lambda): %lf\n", t, lambda, y );
```
```suggestion
		printf( "t: %lf, λ: %lf, M_X(t;λ): %lf\n", t, lambda, y );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/planck/mgf/README.md:
```diff
@@ -138,6 +138,102 @@ for ( i = 0; i < lambda.length; i++ ) {
+        t = random_uniform( 0.0, 10.0 );
+        lambda = random_uniform( 0.1, 5.0 );
+        y = stdlib_base_dists_planck_mgf( t, lambda );
+        printf( "t: %lf, lambda: %lf, M_X(t; lambda): %lf\n", t, lambda, y );
```
```suggestion
        printf( "t: %lf, λ: %lf, M_X(t;λ): %lf\n", t, lambda, y );
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/planck/mgf/README.md:
```diff
@@ -138,6 +138,102 @@ for ( i = 0; i < lambda.length; i++ ) {
+    double t;
+    double lambda;
+    double y;
+    int i;
```
```suggestion
    double lambda;
    double t;
    double y;
    int i;
```

Planeshifter committed:
chore: minor clean-up

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter committed:
style: fix spacing in C benchmark for `planck/mgf`

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter reviewed (approved):
LGTM. Thanks, @yuvi-mittal!

Planeshifter commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into feature/planck-mgf
